### DISCUSSION TITLE
how to avoid launching a duplicate process with task.executor()

### DISCUSSION THREAD

**Discussion by paulius2k** [custom-components/pyscript#267] [Q&A] [answered]
Hello,

I am launching a .py program in HASS with `task.excutor()`.

The issue is that I'd like to check whether the task is already running and not launch a second duplicate process.

Is there a way to check that?

Thank you.

**Comment by craigbarratt** [comment, marked as answer]
You could consider wrapping it in a function that you call with `task.create()`.  That function would call `task.unique()` then task.excutor().  Calling that function multiple times via `task.create()` will have no effect if one already exists.  You could also use `task.name2id()` to check if the task exists.

**Reply by paulius2k** [reply]
thank you @craigbarratt.

I've been trying to do like you suggested for some time with no success.
Could you help me write those few lines?

I have this function, which I run via a service call:

```
@service
def my_test():

	import sys
	if "/config/pyscript_modules" not in sys.path:
		sys.path.append("/config/pyscript_modules")

	import importlib
	import my_scanner
	importlib.reload(my_scanner)
	task.executor(my_scanner.scan,5,1,5)
```

So how would I call it with `task.unique()` and `task.create()` so that it doesn't launch a duplicate?

**Reply by dlashua** [reply]
I have not tested this, but, if that is the entirety of the code you're trying to run, I think you can just add `task.unique('my_scanner')` directly above the call to `task.executor`.

**Reply by paulius2k** [reply]
@dlashua thank you - it worked :)

I used `scan` (the name of my function in `my_scanner.py`) as the `task_name` parameter
```
	[...]
	task.unique('scan', kill_me=True)
	task.executor(my_scanner.scan,5,1,5)
```
